Automattic/ElasticPress
# Update/transient lock to be per site in multisite

* This results in option to run multiple index operations in multisite
* The dashboard indexing was not touched and therefore stays on per
network lock level

## Drawback

In most cases indexing with per-site lock is fine as the documents go to a different ES (ElasticSearch) indicies, but when indexing global indexable (Users) there could be a conflict as those operations touch the same index in ES. We need to take special care of that or introduce another global indexable lock.

## Testing
 As the indexing on small test site is kinda too fast for my window switching skills I end up adding `sleep(10)` in `Command.php:552` (needs to be after `$this->reset_transient();`);
 
* setup an multisite environment with `EP_IS_NETWORK` set
* run `wp vip-search index --url=<url1>` 
* run `wp vip-search index --url=<url1>` again and you should get an error that indexing is ongoing.
* run `wp vip-search index --url=<url2>` make sure the first indexing hasn't finished. This should let you index stuff just fine.
